sqldb/sqlc/queries/invoices.sql

```diff
@@ -81,12 +88,25 @@ ORDER BY
     END DESC
 LIMIT @num_limit OFFSET @num_offset;
 
--- name: UpdateInvoice :exec 
-UPDATE invoices 
-SET preimage=$2, state=$3, amount_paid_msat=$4
-WHERE id=$1;
+-- name: UpdateInvoiceState :execresult
+UPDATE invoices
+SET state = $2,
+    preimage = COALESCE(preimage, $3),
+    settled_index = COALESCE(settled_index, $4),
+    settled_at = COALESCE(settled_at, $5)
+WHERE id = $1;
+
+-- name: UpdateInvoiceAmountPaid :execresult
+UPDATE invoices
+SET amount_paid_msat = $2
+WHERE id = $1;
 
--- name: DeleteInvoice :exec
+-- name: NextInvoiceSettledIndex :one
+UPDATE invoice_sequences SET current_value = current_value + 1
+WHERE name = 'settled_index'
```

Comment:
Wouldn't necessarily need this with the event sourced idea, tho a query like this would be a bit more involved.